packages/components/close-button/src/CloseButton.tsx

```diff
@@ -0,0 +1,58 @@
+import { ButtonStylesProps } from '@spark-ui/button/src/Button.styles'
+import { Icon } from '@spark-ui/icon'
+import { DeleteFill } from '@spark-ui/icons/dist/icons/DeleteFill'
+import React, { forwardRef, PropsWithChildren, useCallback, useRef } from 'react'
+
+import { closeButtonIconStyles, closeButtonStyles } from './CloseButton.styles'
+
+export interface CloseButtonProps
+  extends PropsWithChildren<Omit<React.ButtonHTMLAttributes<HTMLButtonElement>, 'disabled'>>,
+    ButtonStylesProps {
+  disabled?: boolean
+  onClick?: React.MouseEventHandler<HTMLSpanElement>
+  label?: string
+}
+
+export const CloseButton = forwardRef<HTMLSpanElement, PropsWithChildren<CloseButtonProps>>(
+  (
+    { disabled, onClick, className, label = 'close', children = <DeleteFill />, ...props },
+    forwardedRef
+  ) => {
+    const buttonRef = useRef<HTMLDivElement>(null)
+    const onCloseHandler = useCallback(
+      (event: React.MouseEvent<HTMLButtonElement>) => {
+        event.stopPropagation()
+        onClick && onClick(event)
+      },
+      [onClick]
+    )
+
+    const onKeyUpHandler = useCallback((event: React.KeyboardEvent<HTMLDivElement>) => {
+      ;['Enter'].includes(event.key) && buttonRef.current?.click()
+    }, [])
+
+    return (
+      <span
```

Comment:
I am confused about the wrappers of this component: `span > div`.

Why do we need two level ? Why is the div with a role `button` not a `button` (html) ?

In the popover I derivated the close button from an IconButton directly, maybe we can do that ?

Reply:
there is not an iconbutton:

the clicking area for the closing action is bigger than the icon per-se, the componnet is preventing the action to go up cuz it can be placed inside of another button (chip) thats why its a div roled